KaitaiStream.cs

```diff
@@ -619,30 +703,61 @@ public static long Mod(long a, long b)
         }
 
         /// <summary>
-        /// Compares two byte arrays in lexicographical order.
+        /// Compare two byte arrays in lexicographical order. The arrays may not be equal length.
         /// </summary>
-        /// <returns>negative number if a is less than b, <c>0</c> if a is equal to b, positive number if a is greater than b.</returns>
+        /// <returns>Negative number if a is less than b, <c>0</c> if a is equal to b, positive number if a is greater than b.</returns>
         /// <param name="a">First byte array to compare</param>
-        /// <param name="b">Second byte array to compare.</param>
+        /// <param name="b">Second byte array to compare</param>
         public static int ByteArrayCompare(byte[] a, byte[] b)
         {
             if (a == b)
                 return 0;
             int al = a.Length;
             int bl = b.Length;
             int minLen = al < bl ? al : bl;
-            for (int i = 0; i < minLen; i++) {
+
+            for (int i = 0; i < minLen; i++) 
+            {
                 int cmp = a[i] - b[i];
                 if (cmp != 0)
                     return cmp;
             }
 
             // Reached the end of at least one of the arrays
-            if (al == bl) {
-                return 0;
-            } else {
-                return al - bl;
-            }
+            return al == bl ? 0 : al - bl;
+        }
+
+        /// <summary>
+        /// Compare two byte arrays for exact equality.
+        /// </summary>
+        /// <remarks>
+        /// .NET has Array.Equals and == operators, but those check object identity.
+        /// </remarks>
+        public static bool ByteArrayEqual(byte[] a, byte[] b)
+        {
+            int al = a.Length;
+            int bl = b.Length;
+
+            if (al != bl)
+                return false;
+
+            for (int i=0; i<al; i++)
+                if (a[i] != b[i])
+                    return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if byte array is all zeroes.
+        /// </summary>
+        public static bool ByteArrayZero(byte[] a)
```

Comment:
Probably `IsByteArrayZero` would be a better name, as otherwise it kind of clashes with Microsoft's naming of functions like `ZeroMemory`, which actually sets all members of array to zero. Also, I'm not sure that it should be `public`.

Reply:
Changed name to IsByteArrayZero. Fixed. The other helper methods are already public.

Reply:
Other static methods like `Mod`, `ByteArrayCompare`, etc, are public because they are actually invoked in class sources generated by ksc. But then again, `IsByteArrayZero` might be useful to somebody per se...